packages/expo-web-browser/src/WebBrowser.ts

```diff
@@ -140,7 +140,11 @@ async function _openAuthSessionPolyfillAsync(
   try {
     return await Promise.race([openBrowserAsync(startUrl), _waitForRedirectAsync(returnUrl)]);
```

Comment:
In Android, after call ```openBrowserAsync```, it will open the browser and resolve the promise with ```{type: 'opened'}``` immediately.
Because this ```Promise.race([openBrowserAsync(startUrl), _waitForRedirectAsync(returnUrl)])```
it also kill _waitForRedirectAsync immediately and go to ```finally``` and call ```removeEventListener```.

Reply:
cc @mczernek

Reply:
@sjchmiela - looks like this is another issue, working on a fix for it right now